woocommerce/storefront
# Dynamically update block editor layout and styling

**What does it do**

There are two types of types of layout configuration in Storefront: with or without a sidebar.

Storefront is smart enough to automatically adapt its layout. If there are widgets in the main sidebar, then the content will adjust and display the sidebar. Otherwise, when there are no widgets in the sidebar, it will always display the full width layout.

On pages, it's also possible to use the full width template to allow the content to span the full width of the site.

This PR replicates this behaviour in the new blocks editor:

* When the layout is full width, the new alignment options `wide` and `full` are made available.
* If the page/post being edited has a sidebar, `wide` and `full` options are dynamically removed and any existing blocks with this alignment set will have it updated to default.
* Content width changes in the editor to replicate the size of content in the frontend.

**How to test**

1. Add widgets to the sidebar and confirm that they show up in the frontend
2. Create a new page and add some content to it.
3. Apply the 'full width' page template.
4. The size of the content area in the editor should increase.

![editor-full-width](https://user-images.githubusercontent.com/1177726/50293174-dea65d00-046a-11e9-96f0-0bcffb6a4534.gif)

Also:

1. While editing a page with a "full width" page template, add some blocks with wide and full alignment (cover image, columns, etc).
2. No switch back to the "default" page template.
3. The content size should decrease, and blocks set as wide/full should also been updated to default alignment.

![editor-wide](https://user-images.githubusercontent.com/1177726/50293685-1235b700-046c-11e9-917f-0aa174b75c08.gif)

**Other**

* Package.json updated to add babel as a dependency.
* Gruntfile.js updated with new babel task.
* New `assets/js/src/editor.js` is written in ES6 and then when building with Grunt converted to ES5 via babel to `assets/js/editor.js`. This is a bit confusing but allows us to write modern JavaScript while maintaining the current pipeline in Grunt.
